docs/language/learn-ql/cpp/guards.rst

```diff
@@ -0,0 +1,53 @@
+Using the guards library in C and C++
+=====================================
+
+Overview
+--------
+The guards library (defined in ``semmle.code.cpp.controlflow.Guards``) provides a class ``GuardCondition`` representing Boolean values that are used to make control flow decisions. A ``GuardCondition`` is considered to guard a basic block if the block is only reached if the ``GuardCondition`` evaluated a certain way. For instance, in the following code, ``x < 10`` is a ``GuardCondition``, and it guards all the code before the return statement.
+
+.. code:: cpp
+
+    if(x < 10) {
+      f(x);
+    } else if (x < 20) {
+      g(x);
+    } else {
+      h(x);
+    }
+    return 0;
+
+The ``ensuresEq`` and ``ensuresLt`` predicates
+----------------------------------------------
+The ``ensuresEq`` and ``ensuresLt`` predicates are the main way of determining what, if any, guarantees the ``GuardCondition`` provides for a given basic block.
```

Comment:
Perhaps provide links to the standard library references?
https://help.semmle.com/qldoc/cpp/semmle/code/cpp/controlflow/IRGuards.qll/predicate.IRGuards$GuardCondition$ensuresEq.5.html and 
https://help.semmle.com/qldoc/cpp/semmle/code/cpp/controlflow/Guards.qll/predicate.Guards$GuardCondition$ensuresLt.5.html?

Reply:
It might also be helpful to add lower-level headings for each `ensures` and `compares` predicate to break the text up.

Reply:
Added the lower-level headings. I've put a single link to the `GuardCondition` documentation page in the overview section.